### PR TITLE
[BEAM-646] Add PTransformMatcher

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
This interface is used to identify PTransforms to replace during
Pipeline Surgery.

Add ClassPTransformMatcher as the initial simple matcher.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: runners/core-java/src/main/java/org/apache/beam/runners/core/runnerapi/ClassPTransformMatcher.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * transform.
+ *
+ * <p>Once {@link PTransform PTransforms} have URNs, this will be removed and replaced with a
+ * UrnPTransformMatcher.
+ */
+@Experimental(Kind.CORE_RUNNERS_ONLY)
+public class ClassPTransformMatcher implements PTransformMatcher {
+  private final Class<? extends PTransform> clazz;
+
+  public static ClassPTransformMatcher create(Class<? extends PTransform> clazz) {
```
Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/runnerapi/ClassPTransformMatcher.java]: More descriptive name? What is the relationship of the matcher to the class that you pass in?

File: runners/core-java/src/main/java/org/apache/beam/runners/core/runnerapi/package-info.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Provides utilities for interacting with the Beam Runner API.
```
Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/runnerapi/package-info.java]: Just put it top level for now. We'll separate the "stuff for runner shims" from the "stuff for workers" later. I don't necessarily think the terminology we have today is going to stick.

Comment (tgroh) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/runnerapi/package-info.java]: Done.

File: runners/core-java/src/main/java/org/apache/beam/runners/core/runnerapi/ClassPTransformMatcher.java (hunk 1 above)
Comment (tgroh) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/runnerapi/ClassPTransformMatcher.java]: Renamed top level to `PTransformMatchers`, and made this `classEqualTo`

**Pull request closed without merging**